runtime/connectors/s3/s3.go

```diff
@@ -72,53 +71,47 @@ func (c connector) Spec() connectors.Spec {
 	return spec
 }
 
-func (c connector) ConsumeAsFile(ctx context.Context, source *connectors.Source, callback func(filename string) error) error {
+func (c connector) ConsumeAsFile(ctx context.Context, source *connectors.Source) (string, error) {
 	conf, err := ParseConfig(source.Properties)
 	if err != nil {
-		return fmt.Errorf("failed to parse config: %v", err)
+		return "", fmt.Errorf("failed to parse config: %v", err)
 	}
 
 	// The session the S3 Downloader will use
 	sess, err := getAwsSessionConfig(conf)
 	if err != nil {
-		return fmt.Errorf("failed to start session: %v", err)
+		return "", fmt.Errorf("failed to start session: %v", err)
 	}
 
 	// Create a downloader with the session and default options
 	downloader := s3manager.NewDownloader(sess)
 
 	bucket, key, extension, err := getAwsUrlParts(conf.Path)
 	if err != nil {
-		return fmt.Errorf("failed to parse path %s, %v", conf.Path, err)
+		return "", fmt.Errorf("failed to parse path %s, %v", conf.Path, err)
 	}
 
 	f, err := os.CreateTemp(
 		os.TempDir(),
-		fmt.Sprintf("%s*.%s", source.Name, extension),
+		fmt.Sprintf("%s*%s", source.Name, extension),
 	)
 	if err != nil {
-		return fmt.Errorf("os.Create: %v", err)
+		return "", fmt.Errorf("os.Create: %v", err)
 	}
-	defer func() {
-		f.Close()
-		os.Remove(f.Name())
-	}()
+	defer f.Close()
```

Comment:
If the download fails, it looks like there's a code path where `f.Close` gets called twice, which might not be good (unless it just returns an error that we ignore?)